src/components/base/ucc_base_iface.h

```diff
@@ -98,9 +98,16 @@ typedef struct ucc_base_team_iface {
 } ucc_base_team_iface_t;
 
 typedef struct ucc_team ucc_team_t;
+
+enum {
+    UCC_BASE_COLL_ARGS_RADIX
+};
+
 typedef struct ucc_base_coll_args {
+    uint64_t        mask;
     ucc_coll_args_t args;
     ucc_team_t     *team;
+    uint32_t        radix;
```

Comment:
I don't think radix is generic enough to add it to base args. Why not extend tl ucp collective init so that it can take additional argument?

Reply:
I want to use our common interface for tl_ucp_reduce_scatter_knomial and allgather_knomial so that i could use "schedule/task" mechanisms (and also all this should be available for custom schedules in hierarchical CL). If you break the fn signature - it won't work. 

SRA schedule allocates 2 tasks (2 different base_args) and needs to set the radix somehow for those tasks. I also felt it is not ideal place, but couldn't find another way.

Reply:
ok, why not to have two init functions?
```c
ucc_tl_ucp_allgather_init_internal(task, radix)
{

}
ucc_tl_ucp_allgather_init(task)
{
    return ucc_tl_ucp_allgather_init_internal(task, radix_from_config);
}
```

Reply:
yes this is a possibility i considered. The only issue  i had is: if TL/hier would like to use reduce_scatter + allgather (on 1st level of hierarchy for example) with the radix value that TL/hier wants to set - how would this work? looks like it will be option of tl/hier and it will have to use generic TL interface (that has only base args, and task).

Reply:
added internal init function. removed radix from bargs.